### PR TITLE
[TEVA-1790] add radio/checkbox component variants

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-1790

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2739.london.cloudapps.digital>

File: app/components/shared/searchable/collection_checkboxes_component/collection_checkboxes_component.html.slim [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    .collection-component__search
+      input.govuk-input.collection-component__search-input.icon.icon--left.icon--search placeholder="Search"
+      button.collection-component__search-button type="submit"
+
+  = form.govuk_collection_check_boxes attribute_name,
+    collection,
+    value_method,
+    text_method,
+    hint_method,
+    small: @small || @searchable,
```
Comment (cpjmcquillan) [inline app/components/shared/searchable/collection_checkboxes_component/collection_checkboxes_component.html.slim:12]: small and searchable are always the same?

**Comment by cpjmcquillan** [comment]
Instead of having two different components that are essentially exactly the same, I think we could probably have one component and use `.with_variant` to render a different template depending on whether it is checkboxes or radios

File: app/views/publishers/vacancies/build/schools.html.slim [hunk 2]
```diff
@@ -14,22 +14,24 @@
...
-            hint_method: :address))
-
-      - elsif @multiple_schools && @school_options.count <= 10
-        = f.govuk_collection_check_boxes :organisation_ids, @school_options, :id, :name, :address, classes: "checkbox-label__bold"
-
-      / TODO: Implement Shared::SearchableCollectionRadiobuttonsComponent
-
-      - elsif !@multiple_schools
-        = f.govuk_collection_radio_buttons :organisation_ids, @school_options, :id, :name, :address
+      = f.govuk_fieldset legend: { text: t("helpers.legend.publishers_job_listing_schools_form.organisation_ids_html") } do
```
Comment (cesidio) [inline app/views/publishers/vacancies/build/schools.html.slim:17]: aren't we gonna have an extra fieldset legend when the component is not searchable?

Comment (alexbowen) [reply app/views/publishers/vacancies/build/schools.html.slim:17]: i was a bit confused when we paired last week why you mentioned this as when i look at the figma the fieldset legend is always there, although granted the fieldset word `school` becomes `schools` for different scenario which i need to put into this PR. https://www.figma.com/file/wyEeHQ1U9aCXOJdzh8lF8a/Sprint-72?node-id=91%3A1095 1.1.1 1.2.2

File: app/components/shared/searchable_collection_component.rb [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
     @collection = collection
     @value_method = value_method
     @text_method = text_method
     @hint_method = hint_method
+
+    @small = searchable
+  end
+
+  def searchable
+    @collection.count >= @threshold
```
Comment (cpjmcquillan) [inline app/components/shared/searchable_collection_component.rb:17]: if we're using `attr_accessor` you can drop the `@`

File: app/components/shared/searchable_collection_component.rb [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
     @collection = collection
     @value_method = value_method
     @text_method = text_method
     @hint_method = hint_method
+
+    @small = searchable
+  end
+
+  def searchable
+    collection.count >= @threshold
```
Comment (david-mears-dfe) [inline app/components/shared/searchable_collection_component.rb:17]: It's good Ruby hygiene to use an attr_reader for reading instance variables that you aren't intending to write to. This applies to collection and threshold.

**Comment by david-mears-dfe** [comment]
Can you provide screenshots, as the orgs available on the review app don't have >10 schools?

File: app/components/shared/searchable_collection_component/searchable_collection_component.html+radiobutton.slim [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    .collection-component__search
+      input.govuk-input.collection-component__search-input.icon.icon--left.icon--search placeholder="Search"
+      button.collection-component__search-button type="submit"
+
+  = form.govuk_collection_radio_buttons attribute_name,
+    collection,
+    value_method,
+    text_method,
+    hint_method,
+    small: @small,
```
Comment (cpjmcquillan) [inline app/components/shared/searchable_collection_component/searchable_collection_component.html+radiobutton.slim:12]: we could attr_accessor this if we are for everything else

File: app/components/shared/searchable_collection_component/searchable_collection_component.html+checkbox.slim [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    .collection-component__search
+      input.govuk-input.collection-component__search-input.icon.icon--left.icon--search placeholder="Search"
+      button.collection-component__search-button type="submit"
+
+  = form.govuk_collection_check_boxes attribute_name,
+    collection,
+    value_method,
+    text_method,
+    hint_method,
+    small: @small,
```
Comment (cpjmcquillan) [inline app/components/shared/searchable_collection_component/searchable_collection_component.html+checkbox.slim:12]: we could attr_accessor this if we are for everything else

**Review by cpjmcquillan — APPROVED** [review]
LGTM (pending the padding issue I discussed with @alexbowen)

**Pull request merged**